pages/api/unsubscribe.js

```diff
@@ -0,0 +1,86 @@
+import { connectToDatabase } from "../../lib/mongodb/connect";
+import { unsubscribeUser } from "../../lib/users/unsubscribeUser";
+import { getUserByEmail } from "../../lib/users/getUser";
+import { submitEmail } from "../../lib/notify/submitEmail";
+
+export default async function handler(req, res) {
+  if (req.method === "POST") {
+    let data = req.body;
+    let userObj;
+    const origin = req.headers.origin;
+    if (data.email) {
+      const conn = await connectToDatabase(
+        process.env.MONGO_URL,
+        process.env.MONGO_DB
+      );
+      try {
+        userObj = await getUserByEmail(conn.db, data.email);
+      } catch (e) {
+        console.log(e);
+        return res.redirect("/error");
+      }
+
+      if (userObj) {
```

Comment:
If the email does not exist in the db, (I assume it returns null or undefined), would we want to return an error message? ``` else {
        return res.status(400).end("Email not found");
      } ```

Reply:
That is the right idea, but since this is an api route, that would just load a blank screen with the text `Email not found`. The `/error` route takes query string parameters for custom error messages and displays them in a similar fashion as the 404 and 500 error pages.